internal/pkg/db/memory/addressables_test.go

```diff
@@ -0,0 +1,737 @@
+
+package memory
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/edgexfoundry/edgex-go/internal/pkg/db"
+
+	contract "github.com/edgexfoundry/go-mod-core-contracts/models"
+
+	"github.com/google/uuid"
+)
+
+// allTestAddressables a slice containing all the pre-constructed addressables for testing.
+var allTestAddressables = []contract.Addressable{addressable1, addressable2, addressable3}
+
+// addressable1 is a pre-constructed addressable with unique properties which does not match any other addressable when
+//  invoking functions which can return multiple addressables such as GetAddressablesByPort
+var addressable1 = contract.Addressable{
+	Timestamps: contract.Timestamps{
+		Created:  123,
+		Modified: 456,
+		Origin:   789,
+	},
+	// This is a valid UUID which allows us to ensure that the ID used is not replaced.
+	Id:         "6d40520f-035b-4be9-bacc-152d4f4f37a9",
+	Name:       "FirstAddressable",
+	Protocol:   "tcp",
+	HTTPMethod: "POST",
+	Address:    "0.0.0.0",
+	Port:       8000,
+	Path:       "/",
+	Publisher:  "Mushroom Kingdom",
+	User:       "Yoshi",
+	Password:   "MarioIsGettingHeavy",
+	Topic:      "dinosaur-things",
+}
+
+// addressable2 is a pre-constructed addressable with non-unique properties. The properties which can lead to multiple
+// results when invoking the client, such as GetAddressablesByPort, have matching properties with addressable3. This
+// helps with testing situations where multiple addressables are returned
+var addressable2 = contract.Addressable{
+	Timestamps: contract.Timestamps{
+		Created:  789,
+		Modified: 456,
+		Origin:   123,
+	},
+	// This is a valid UUID which allows us to ensure that the ID used is not replaced.
+	Id:         "1d874125-df04-4718-bd81-b550019b74aa",
+	Name:       "SecondAddressable",
+	Protocol:   "http",
+	HTTPMethod: "GET",
+	Address:    "127.0.0.1",
+	Port:       8080,
+	Path:       "/home",
+	Publisher:  "Nintendo",
+	User:       "Link the hero time",
+	Password:   "RescueZelda",
+	Topic:      "hyrule-events",
+}
+
+// addressable3 is a pre-constructed addressable with non-unique properties. The properties which can lead to multiple
+// results when invoking the client, such as GetAddressablesByPort, have matching properties with addressable2. This
+// helps with testing situations where multiple addressables are returned
+var addressable3 = contract.Addressable{
+	Timestamps: contract.Timestamps{
+		Created:  789,
+		Modified: 456,
+		Origin:   123,
+	},
+	// This is a valid UUID which allows us to ensure that the ID used is not replaced.
+	Id:         "cb7d56ca-527d-448b-91f0-bc27f9891b90",
+	Name:       "ThirdAddressable",
+	Protocol:   "stomp",
+	HTTPMethod: "PUT",
+	Address:    "127.0.0.1",
+	Port:       8080,
+	Path:       "/castle",
+	Publisher:  "Nintendo",
+	User:       "Gannon",
+	Password:   "GetTriforce",
+	Topic:      "hyrule-events",
+}
+
+func TestClient_AddAddressableUUID(t *testing.T) {
+	client := NewClient()
```

Comment:
Create a private `newSUT` method that returns an instance of the system under test -- in this case, the result of `NewClient`.  Call `newSUT` from each of your tests.